File: library/src/main/java/com/sideslip/view/ItemType.java
package com.sideslip.view;

import com.sideslip.face.IItemType;


/**
 * created by wyu on 2021/3/6.
 */



public abstract class ItemType<T> implements IItemType {


    @Override
    public final int type(Object data, int position) {
        return getItemOrder((T) data, position);
    }

    public abstract int getItemOrder(T data,int position);
}

File: library/src/main/java/com/sideslip/view/ItemType.kt
package com.sideslip.view

import com.sideslip.face.IItemType

/**
 * created by wyu on 2021/3/6.
 */
abstract class ItemType<T> : IItemType {
    override fun type(data: Any, position: Int): Int {
        return getItemOrder(data as T, position)
    }

    abstract fun getItemOrder(data: T, position: Int): Int
}